src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -824,10 +831,13 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
+            });
+            return Constants.S_OK;
```

Comment:
S_OK says that we are already stopped. Generally we should be returning S_ASYNC_STOP. A complete implementation of this stuff is complicated. I would probably suggest adding a --
// TODO: this should be returning S_ASYNC_STOP
for now and fixing this in a subsequent checkin.